pkg/master/master.go

```diff
@@ -96,10 +96,53 @@ const (
 	DefaultEtcdPathPrefix = "/registry"
 )
 
+type DatabaseClients struct {
+	Default      storage.Interface
+	Experimental storage.Interface
+	Overwrites   map[string]storage.Interface
+}
+
+func (d *DatabaseClients) get(resource string, experimental bool) storage.Interface {
+	if d.Overwrites != nil {
+		if client, ok := d.Overwrites[resource]; ok {
+			return client
+		}
+	}
+	if experimental {
+		return d.Experimental
+	}
+	return d.Default
+}
+
+func (d *DatabaseClients) backends() []string {
+	backends := map[string]struct{}{}
+	if d.Default != nil {
+		for _, backend := range d.Default.Backends() {
+			backends[backend] = struct{}{}
+		}
+	}
+	if d.Experimental != nil {
+		for _, backend := range d.Experimental.Backends() {
+			backends[backend] = struct{}{}
+		}
+	}
+	if d.Overwrites != nil {
+		for _, storage := range d.Overwrites {
+			for _, backend := range storage.Backends() {
+				backends[backend] = struct{}{}
+			}
+		}
+	}
+	var result []string
+	for backend := range backends {
+		result = append(result, backend)
```

Comment:
I guess I don't understand this function-- I can't imagine how it could possibly be valid to put backends from different storage destinations in the same list?